Source/DafnyCore/Verifier/Translator.cs

```diff
@@ -1063,13 +1064,28 @@ internal class PredefinedDecls {
                   Bpl.Expr startFuelAssert_expr = new Bpl.IdentifierExpr(f.tok, startFuelAssert);
                   this.functionFuel.Add(new FuelConstant(f, baseFuel_expr, startFuel_expr, startFuelAssert_expr));
                 }
+
+                if (f.IsOpaque) {
+                  MaybeCreateRevealableConstant(f);
+                }
               }
             }
           }
         }
       }
     }
 
+    private void MaybeCreateRevealableConstant(Function f) {
```

Comment:
The "`Maybe`" part of the name suggests to me that it may or may not be appropriate to have a revealable constant at all. But here it just indicates that the operation is idempotent--it's performed only once. For that purpose, I suggest still just naming the method `CreateRevealableConstant`.